### PR TITLE
Recognize characters `,=[]` in google-style docstrings

### PR DESCRIPTION
**Pull request by jmatzdorff-cpi**
Fixes https://github.com/pdoc3/pdoc/issues/3

following add to the tests for verification:
```
            arg3 (str, optional): Description of arg3
            arg4 (str, optional, default=10): Description of arg4
            arg5 (str,optional,default=10): Description of arg5
            arg6 (str or int,optional,default = 10): Description of arg6
```

### CODE REVIEW

**Comment by jmatzdorff-cpi** [comment]
drat, conflicts.  i didn't bother to pull and merge.  i'll do that in a bit and repush.

File: pdoc/test/example_pkg/__init__.py [hunk 1]
```diff
@@ -151,25 +151,26 @@ def google(self):
...
 
-                test_sequence =
-                code
+                  test_sequence =
+                  code
 
-            Continue.
-          *args: passed around
+              Continue.
+            *args: passed around
```
Comment (kernc) [inline pdoc/test/example_pkg/__init__.py]: Make a new `Args:` section for your tests. This 2-spaces indented section was covering https://github.com/pdoc3/pdoc/issues/5.

File: pdoc/test/__init__.py [hunk 2]
```diff
@@ -660,6 +660,14 @@ def test_google(self):
 <dd>Description of arg1</dd>
 <dt><strong><code>arg2</code></strong> :&ensp;<code>str</code> or <code>int</code></dt>
 <dd>Description of arg2</dd>
+<dt><strong><code>arg3</code></strong> :&ensp;<code>str</code>, optional</dt>
+<dd>Description of arg3</dd>
+<dt><strong><code>arg4</code></strong> :&ensp;<code>str</code>, <code>optional</code>, <code>default</code>=<code>10</code></dt>
```
Comment (kernc) [inline pdoc/test/__init__.py]: You get flake8 to ignore long lines by putting `# noqa: E501` at the end of the string literal.

But in this case, we actually have another issue. This line should be:
```html
<dt><strong><code>arg4</code></strong> :&ensp;<code>str</code>, optional, default=<code>10</code></dt>
```
Do you want to have a look at `_ToMarkdown._deflist()`'s regular expressions and somehow help it skip over `r'\bdefault[= ]'` as well?
https://github.com/pdoc3/pdoc/blob/7500791bef2ff61c4d8bab093b6e2b6ece446836/pdoc/html_helpers.py#L108-L125

Comment (jmatzdorff-cpi) [reply pdoc/test/__init__.py]: done

File: pdoc/test/example_pkg/__init__.py (hunk 1 above)
Comment (jmatzdorff-cpi) [reply pdoc/test/example_pkg/__init__.py]: done

File: pdoc/test/__init__.py (hunk 2 above)
Comment (kernc) [reply pdoc/test/__init__.py]: No, no, exactly the opposite. Optional and default are _not_ to be wrapped in `<code>`. They are surrounding text matter, not _literal code_ or to-be-linked _identifiers_.  :grimacing:

**Comment by kernc** [comment]
I dared fix it myself. In fact, managed to simplify it to a much higher level of comprehensibility. :grin:

Thank you!

**Pull request merged**